### ISSUE TITLE
Can't add data to tables.

### ISSUE THREAD

**Issue by snowblizz** [tabulate/tabulate#54]
Hi, I installed Tabulate and after massaging some data for an import finally realised I can't add any data to tables I have created with Tabulate.
I get the following error message.

'c80aps1.wp_tabulate_changesets' doesn't exist

Running version 4.8 of WP.

**Comment by samwilson** [comment]
I'm sorry for this error.

It sounds like something went wrong with the activation of Tabulate. Could you try deactivating and then reactivating it? The required database tables should be created.

**Comment by snowblizz** [comment]
Hi,

I've tried deactivating and then activating, no luck. I tried uninstalling and deleting the old tables, reinstalling plugin and no luck.

I "originally" installed the 2.9.5 update without deactivating the plugin though. I think saw something relating to such issues, can that mess with somethign somewhere that's ghost in the machine maybe?

Also, ofc WP insists plugin is "untested with your version of WP", not sure if that has any bearing.

Can it be a deeper permissions issue? I sort of inherited authorship of the WP site am working with. I should have all back end access but I don't trust the competence of the ISP fully either.

**Comment by samwilson** [comment]
(Sorry for the delay in replying.)

Tabulate checks for the existence of its database tables when it's activated, and creates them if required. So this is a tricky one to debug!

It must be failing to create the changesets table, and failing to report that failure.

This could *possibly* be related to the storage engine(s) in use in your database. Is the default something other than InnoDB?

**Comment by snowblizz** [comment]
Having finally remember to check this when I am at place where I can access the database itself it seems the database is running on MariaDB (10.0.31-MariaDB-0ubuntu0.16.04.2 - Ubuntu 16.04, to be precise, if that makes more sense to you than it did me).
I tried it on a "clean install" on subsite and had same issue so not only a bodged install/update it seems.

**Comment by samwilson** [comment]
Can you provide details about the storage engines you're using? e.g. check in phpMyAdmin, or get the output of `SHOW ENGINES;` and `SELECT engine FROM information_schema.TABLES WHERE TABLE_SCHEMA = 'dbname';`

Sorry about this error. Would love to get it fixed; thanks for helping!

**Comment by snowblizz** [comment]
show engines gives me:
`
Engine | Support | Comment | Transactions | XA | Savepoints |  
-- | -- | -- | -- | -- | -- | --
MyISAM | YES | MyISAM storage engine | NO | NO | NO
CSV | YES | CSV storage engine | NO | NO | NO
MEMORY | YES | Hash based, stored in memory, useful for temporary... | NO | NO | NO
BLACKHOLE | YES | /dev/null storage engine (anything you write to it... | NO | NO | NO
FEDERATED | YES | FederatedX pluggable storage engine | YES | NO | YES
MRG_MyISAM | YES | Collection of identical MyISAM tables | NO | NO | NO
ARCHIVE | YES | Archive storage engine | NO | NO | NO
InnoDB | DEFAULT | Percona-XtraDB, Supports transactions, row-level l... | YES | YES | YES
PERFORMANCE_SCHEMA | YES | Performance Schema | NO | NO | NO
Aria | YES | Crash-safe tables with MyISAM heritage | NO | NO | NO
`
and the other one gives me:
`

engine |  
-- | --
InnoDB
InnoDB
MyISAM
MyISAM
MyISAM
MyISAM
MyISAM
MyISAM
InnoDB
InnoDB
InnoDB
MyISAM
MyISAM
InnoDB
MyISAM
MyISAM
MyISAM

`

**Comment by samwilson** [comment]
Yes, it seems that the issue is that the tables being created as InnoDB (your default), but some existing ones (probably `users` is the crux) are using MyISAM. Can you try changing at least the `users` table to InnoDB; there's some interesting points [here](https://wordpress.stackexchange.com/a/15377/58142) about other reasons why this might be a good idea.

**Comment by samwilson** [comment]
(I'll add something to Tabulate that checks and notifies about this issue, if this is what's causing the problems.)

**Comment by tfindlay99** [comment]
I'm getting the "wp_tabulate_changesets' doesn't exist" error. I created a two field table in Tabulate and saved it. I then went to the Create tab to create a record. I filled in the two fields and clicked the button to save the record but the error came up.

**Comment by samwilson** [comment]
@tfindlay99 can you check what storage engines your tables are using please? (Using phpMyAdmin or similar.) It sounds like there is an issue when the `users` table is not InnoDB. I'm preparing a fix for this, but actually I'm not 100% sure yet that this is what the problem is here, so it'd be good to get some more perspectives. Thanks!

**Comment by dcsoliday** [comment]
We're having the same problem. We found this first: https://github.com/tabulate/tabulate/issues/47, which closely matches our situation, but we're on v2.10.3 of the plugin. It's a multisite server, running WP v4.9.7 (until next Tuesday.) The backend is MariaDB  libmysql - 5.5.56-MariaDB.

We've tried activating the plugin on the site and network activating it. It creates these tables, where 5 is the site ID we're trying to use it on:
wp_5_tabulate_reports
wp_5_tabulate_report_sources
wp_tabulate_changes
wp_tabulate_changesets
wp_tabulate_reports
wp_tabulate_report_sources

Our SHOW ENGINES; results:


CSV | YES | CSV storage engine | NO | NO | NO
-- | -- | -- | -- | -- | --
MRG_MYISAM | YES | Collection of identical MyISAM tables | NO | NO | NO
MEMORY | YES | Hash based, stored in memory, useful for temporary... | NO | NO | NO
BLACKHOLE | YES | /dev/null storage engine (anything you write to it... | NO | NO | NO
MyISAM | YES | MyISAM storage engine | NO | NO | NO
InnoDB | DEFAULT | Percona-XtraDB, Supports transactions, row-level l... | YES | YES | YES
ARCHIVE | YES | Archive storage engine | NO | NO | NO
FEDERATED | YES | FederatedX pluggable storage engine | YES | NO | YES
PERFORMANCE_SCHEMA | YES | Performance Schema | NO | NO | NO
Aria | YES | Crash-safe tables with MyISAM heritage | NO | NO | NO

but SELECT engine FROM information_schema.TABLES WHERE TABLE_SCHEMA = 'dbname' comes up empty because there's no TABLES table in the information_schema db.
